### PR TITLE
Extended backend testing suite

### LINKED ISSUE #192: Backend Testing
As a developer I would like to have existing tests which cover all the backend functionality. After establishing a testing basis, we can now build upon this and create multiple more tests.

---

## :white_check_mark: &nbsp; Conditions of satisfaction

- [ ] Should cover all routes with tests and cover possible edge-cases
- [ ] Should attempt to call the routes with malformed and/or wrong data to check for error handling

### PR DESCRIPTION
**Pull request by SanJSp**
Fixes #192 

## Changes
This PR introduces model validation tests for each model and extends the route tests. Furthermore, I've marked routes that are currently not used anymore for a future refactoring issue. I've introduced Chai, as it's needed for the model validation tests.

I've tried to reduce magic numbers and implicit knowledge as much as possible, please still comment as soon as something is not clear.

## Screenshots
The current coverage including the currently unused routes (for which I have not written tests)
grafik
But as soon those get removed, the test coverage looks like this:
grafik
The low branch coverage in `socket` is because of the try catch blocks.

## Checklist before merge

**Developer's responsibilities**
* [ ] **Assign** one or two developers
* [ ] **Change code** if reviewer(s) has/have requested it
* [ ] **Pull request build** has passed
* [ ] **tested locally** (in at least chrome & firefox if frontend)
* [ ] updated the **documentation**
* [ ] added **tests** where necessary

### CODE REVIEW

File: server/controllers/rpaFrameworkCommandsController.js [hunk 1]
```diff
@@ -47,12 +47,13 @@ exports.getVarsForTask = async (req, res) => {
           Task: task,
         },
         {
-          InputVars: 1,
-          Output: 1
+          inputVars: 1,
+          outputValue: 1,
```
Comment (SanJSp) [inline server/controllers/rpaFrameworkCommandsController.js:51]: These changes were necessary to adapt the call to our databse schema, so it can be tested with data similiar to the Db

File: server/models/rpaTaskModel.test.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+/* eslint-disable no-undef */
+const mongoose = require('mongoose');
+const { expect } = require('chai');
+const dbHandler = require('../utils/TestingUtils/testDatabaseHandler.js');
```
Comment (lukashueller) [inline server/models/rpaTaskModel.test.js:4]: There is a small typo: the `TestDatabaseHandler.js` starts with an uppercase Letter. Then the path should be resolveable.
```suggestion
const dbHandler = require('../utils/TestingUtils/TestDatabaseHandler.js');
```

File: server/models/rpaTaskModel.test.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+const { testRpaTask1 } = require('../utils/TestingUtils/testData.js');
+const taskModel = require('./rpaTaskModel.js');
```
Comment (lukashueller) [inline server/models/rpaTaskModel.test.js:6]: Did we need this `taskModel`? ESLint throws an error. So please use `disable no-unused-vars` for this line if it is really needed.

File: server/models/rpaTaskModel.test.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    task.save((err) => {
+      expect(err.errors.Task).to.exist;
+      expect(err.errors.Task.message).equal('Task required');
+    });
+  });
+
+  it('should be invalid if Code is empty', async () => {
+    task.save((err) => {
+      expect(err.errors.Code).to.exist;
+      expect(err.errors.Code.message).equal('Code required');
```
Comment (lukashueller) [inline server/models/rpaTaskModel.test.js:53]: ESLint expects return values for each of your arrow functions. I'm not that deep into the testing syntax but maybe we can fix this warning by directly returning the expression (if there is only one expression in the arrow function body) or return both expressions concatenated with `&&`.

```suggestion
  it('should throw no errors for correct job', async () => {
    task.save((err) => expect(err).to.not.exist);
  });
```

File: server/models/singleSourceOfTruthModel.test.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
+/* eslint-disable no-undef */
+const mongoose = require('mongoose');
+const { expect } = require('chai');
+const dbHandler = require('../utils/TestingUtils/testDatabaseHandler.js');
```
Comment (lukashueller) [inline server/models/singleSourceOfTruthModel.test.js:4]: again the corresponding js-File starts with an uppercase

**Review by lukashueller — CHANGES_REQUESTED** [review]
Hello Sandro,
for me a really great work! Not only the coverage shows how deep you have dealt with the subject! From what I can evaluate the tests are really extensive! This is a huge step for our project!

I have added a few small notes that relate only to ESLint Warnings. Please look over them again and feel free to give feedback if there are some things that can't be fixed. Then I'm also fine with deactivating the corresponding ESLint rule.

File: server/models/rpaTaskModel.test.js (hunk 3 above)
Comment (SanJSp) [reply server/models/rpaTaskModel.test.js:7]: yes, it's needed. Disabled it there

**Review by lukashueller — APPROVED** [review]
Looks fine for me!

File: server/routes/ssot.js [hunk 6]
```diff
@@ -9,8 +9,8 @@ const variableController = require('../controllers/ssotVariableController');
 const rpaAttributesController = require('../controllers/ssotRpaAttributes');
 
 /// PARSING ROUTES ///
-router.get('/parser/get-robot-code', parsingController.getRobotCode);
-router.get('/parser/getForId/:botId', parsingController.getRobotCodeForId);
+router.get('/parser/get-robot-code', parsingController.getRobotCode); // not used anymore
```
Comment (WolfgangDaniel) [inline server/routes/ssot.js:12]: why not delete all the routes that aren't used anymore?

**Review by WolfgangDaniel — APPROVED** [review]
Looks very good! I just have that one question. If there is a good reason for keeping all the roots you can go ahead and merge :)

File: server/routes/ssot.js (hunk 6 above)
Comment (SanJSp) [reply server/routes/ssot.js:12]: Because they are called at other places and this would be a refactoring that would expand over the user story time and bloat the pr.
